### ISSUE TITLE
Reformat code breaks new pure functions shortcut  |-> 

### ISSUE THREAD

**Issue by gdelfino** [halirutan/Mathematica-IntelliJ-Plugin#189]
**Describe the bug**
Since WL 12.2 we can write pure functions like this:

    x |-> x+1

when this code is reformatted, a space is inserted after the vertical bar effectively introducing a bug in the code:

    x | -> x+1

**To Reproduce**
Steps to reproduce the behaviour:
1. Select code that includes a pure function with the new notation
2. Select "Format Code"  from the "Code" menu or press ALT-CTRL-L or 

**Expected behaviour**
The sequence of three characters: |-> should be treated as a single character.  In fact is shown as a single character with fonts such as JetBrains Mono when ligatures are enabled. 

**Please also provide:**
 - OS:  Windows 10
 - IntelliJ IDEA 2021.2.1
 - Wolfram Plugin Version 2021.2-beta-1

**Comment by halirutan** [comment]
Yes, the new operators and syntax are still not incorporated into the parser which leads to operators not being recognized correctly. I have access to the latest Mathematica versions now and can start fixing things up.

Related: #190, #191
